### PR TITLE
Lua_api.txt: Document blockpos, coordinate conversion, map terminology. Fix and improve LVM 'calc_lighting()' warning

### LINKED ISSUE #8356: Document 'blockpos' in lua_api.txt.
Reminder for myself to do this sometime. Helpful for #8140 
```
node_min = blockpos_min * MAP_BLOCKSIZE;
node_max = (blockpos_max + v3s16(1, 1, 1)) * MAP_BLOCKSIZE - v3s16(1, 1, 1);
```
So mapblock at blockpos (0, 0, 0) extends from nodepos (0, 0, 0) to nodepos (15, 15, 15).
```
local BLOCKSIZE = core.MAP_BLOCKSIZE
local function get_blockpos(pos)
	return {
		x = math.floor(pos.x/BLOCKSIZE),
		y = math.floor(pos.y/BLOCKSIZE),
		z = math.floor(pos.z/BLOCKSIZE)}
end
```
Nodepos -> blockpos:
blockposcomp = math.floor(nodeposcomp / 16)

Blockpos -> nodepos:
nodeposblockminp = blockpos * 16
nodeposblockmaxp = blockpos * 16 + 15

### PR DESCRIPTION
**Pull request by paramat**
Closes #8356 and attends to https://github.com/minetest/minetest/issues/8220#issuecomment-472264807

Initially intended to close 8356 but it seemed logical to document coordinates, mapblocks, mapchunks etc. too, so i ended up creating a new lua_api section for this information.

@Ezhh since this is your speciality.

### CODE REVIEW

**Comment by paramat** [comment]
Updated with some improvements to text and MD formatting, looks like:

Screenshot from 2019-03-14 23-19-01

File: doc/lua_api.txt [hunk 1]
```diff
@@ -1189,7 +1189,7 @@ clients and handled by many parts of the engine.
 A 'mapchunk' (sometimes abbreviated to 'chunk') is usually 5x5x5 mapblocks
 (80x80x80 nodes) and is the volume of world generated in one operation by
 map generation.
```
Comment (ClobberXD) [inline doc/lua_api.txt]: `... generated in one pass by the mapgen.`

(Probably relevant, as this line gets touched as well.)

Comment (paramat) [reply doc/lua_api.txt]: That's ok but i think 'operation' is clearer.

Comment (ClobberXD) [reply doc/lua_api.txt]: Sorry for being so vague, I was mainly suggesting to change "map generation" to "mapgen" :)

Comment (paramat) [reply doc/lua_api.txt]: Ahh. I have been wondering about that, i was thinking 'map generation' is clearer than 'mapgen', but 'the mapgen' refers to the core mapgen being used, which may be clearer.

Comment (paramat) [reply doc/lua_api.txt]: Done.

**Comment by Ezhh** [comment]
Sorry for slow response - but just wanted to add that it's good to see this info included. :)

**Pull request merged**